### ISSUE TITLE
Support Integrated Active Directory Password Authentication

### ISSUE THREAD

**Issue by maxali** [Microsoft/vscode-mssql#435]
Cannot CONNECT to Azure server using Active Directory Password Authentication.

- MSSQL Extension Version: 0.1.5
- VSCode Version: 1.7.2
- OS Version: Windows 10

Steps to Reproduce:

1. Create Azure SQL Database
2. Add a user from Azure AD `CREATE USER [your.user@contoso.com] FROM EXTERNAL PROVIDER;`
3. Connect using Active Directory Password Authentication

**Comment by kevcunnane** [comment]
@maxali, thanks for raising this. Currently this is blocked on our upstream dependency on .Net Core, which does not currently have Azure AD support for SQL Client. I have updated issue https://github.com/dotnet/corefx/issues/8807 with this request to ensure the ask is tracked there too. 

If this is fixed in .Net Core there would be additional work to support Azure authentication in general from our application, and it's likely we'd do this and other Azure-related support such as firewall rule creation in a larger improvement.

**Comment by erickangMSFT** [comment]
I assigned it to myself to track the progress with .NET core.

**Comment by michaelaird** [comment]
Adding my $.02:

This should support Active Directory Universal Authentication the way SSMS does. ie. by opening the Azure authentication dialog. This is necessary for enabling those of us that have multi-factor authentication turned on for our AD accounts.

**Comment by anujb** [comment]
+1

It seems lackadaisical that [MFA via] AAD is clearly marketed as differentiated capability, but not supported in our 1st party developer tools for VS Code.

**Comment by AronT-TLV** [comment]
The block is over. Azure AD support is now available in Azure Data Studio (even on Mac) for example

**Comment by SeaDude** [comment]
Is this going to be implemented in MSSQL Extension? I'd like to login using AD Creds rather than SQL Server Creds.

**Comment by Trestomagnetic** [comment]
Any movement on this? My company doesn't allow us to use sql creds, we are AAD - Universal with MFA only. I would love to use VS Code instead of SSMS, but this is a blocker.

**Comment by Msjronkko** [comment]
Any updates on this?

**Comment by ransagy** [comment]
Why is there a difference here between Azure Data Studio? They don't use the same infra in this context?

**Comment by 7even7** [comment]
Azure Data Studio has supported MFA for a long time, and so has SSMS. Why is it still not available in VS code mssql?
VS code with git lense is my preferred environment, but now I'm forced to jump around in different editors/IDEs because of this issue.

**Comment by JeffBor** [comment]
FYI, the Feb 2020 release of ADS supports better token authentication: https://github.com/microsoft/azuredatastudio/issues/3669

**Comment by acambitsis** [comment]
Any update on this?  I am really keen to use this extension in VS Code, but have to keep reverting to Azure Data Studio because of AD / MFA support.

**Comment by ldsalomone** [comment]
Following this. Wondering if this is in development or on hold?

**Comment by vikaspoddar** [comment]
hello  @kburtram,  has the MFA support added to `vscode-mssql`
